src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+    }
+
+    private final HashMap<String, Server> servers = new HashMap<>();
+
+    /**
+     * Check if the given Member has administrator permissions. Notifies the user if they do not have permission.
+     * @param member The Member to check
+     * @return True if the Member has admin, false otherwise
+     */
+    private static boolean checkAdmin(Member member, TextChannel channel) {
+        if (!member.hasPermission(Permission.ADMINISTRATOR)) {
+            channel.sendMessage(member.getAsMention() + " You must have administrator permission to run this command.").queue();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Check if the given Topic exists in the Server. Notifies the user if the topic does not exist.
+     * @param member The Member to reply to
+     * @param channel The Channel to reply within
+     * @param server The Server to check in
+     * @param topic The Topic to check for
+     * @return Contains the Topic, if it exists
+     */
+    private static Optional<Topic> checkTopicExists(Member member, TextChannel channel, Server server, String topic) {
+        Optional<Topic> topicOptional = Arrays.stream(server.getTopics())
+                .filter(t -> t.getName().equalsIgnoreCase(topic))
+                .findFirst();
+        if (topicOptional.isEmpty()) {
+            channel.sendMessage(String.format(
+                    "%s Topic \"%s\" does not exist.",
+                    member.getAsMention(),
+                    topic)).queue();
+        }
+        return topicOptional;
+    }
+
+    @Override
+    public void onReady(@NotNull ReadyEvent event) {
+        System.out.println("Logged in");
+        event.getJDA().getPresence().setPresence(Activity.playing("$help"), false);
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull GuildMessageReceivedEvent event) {
+        String[] tokens = event.getMessage().getContentDisplay().split(" ");
+        if (!tokens[0].startsWith("$")) return;
+
+        String guildID = event.getGuild().getId();
+        Server server = servers.computeIfAbsent(guildID, k -> Server.load(event.getGuild()));
+
+        // pick the correct method to call
+        CommandHandler commandHandler;
+        switch (tokens[0].substring(1)) {
+            case "help" -> commandHandler = this::help;
+            case "maketopic" -> commandHandler = this::maketopic;
+            case "deletetopic" -> commandHandler = this::deletetopic;
+            case "showtopics" -> commandHandler = this::showtopics;
+            case "queue" -> commandHandler = this::queue;
+            case "ready" -> commandHandler = this::ready;
+            case "showqueue" -> commandHandler = this::showqueue;
+            case "clear" -> commandHandler = this::clear;
+            default -> commandHandler = this::unknownCommand;
+        }
+        commandHandler.handle(
+                Objects.requireNonNull(event.getMember()),
```

Comment:
This throws `NullPointerException` if `event.getMember()` is null. I only added this on IntelliJ's suggestion; I think `event.getMember()` only returns `null` for a Discord-generated join/leave message, but those will probably never be detected as a bot command.

Reply:
It is not caught anywhere else, but like I said, I don't think this can ever be caused by an actual user's message, so I don't see any point in showing an error message.